Add identity keys to push (#502)
WalletConnect/walletconnect-docs: docs/specs/clients/push/push-proposal.md

Proposal protocol will be established as follows:

1. Dapp sends push proposal on pairing P with public key X, relay and metadata

> 2. Dapp subscribes to response topic, which is sha256 of a public key X

Reviewer: Could you pls outline why we need a separate response topic now vs responding on pairing topic as we do currently?

Reviewer: ```suggestion
2. Dapp subscribes to response topic, which is the sha256 hash of public key X
```

Author: That's a good question and I am not convinced here what is the best approach. Responding on a pairing topic would require to use pairing's symkey but we want to use envelope type1 to secure the the jwt with a key that has been established by a key exchange.

what would you suggest?

> 2. Dapp subscribes to response topic, which is the sha256 hash of public key X

Author: @bkrem your comment has disappeared after committing your suggestion:
>Could you pls outline why we need a separate response topic now vs responding on pairing topic as we do currently?
so, this is a very good question and I am not convinced here what is the best approach. But using pairing topic would require to use pairing symKey but we want to avoid that and respond with type one envelope to hide a jwt.

Author: on the other hand I see, it's getting ugly with so many topics

3. Wallet receives push proposal with public key X on pairing P
4. Wallet generates key pair Y
5. Wallet derives symmetric key with keys X and Y
6. Push topic is derived from sha256 hash of symmetric key 
7. Wallet subscribes to push topic 
8. Wallet sends proposal response(type1 envelope) on response topic with subscriptionAuth
9. Dapp receives proposal response on response topic
10. Dapp derives symmetric key and decrypts subscriptionAuth

> 11. Dapp registers address at the Cast Server. 

Reviewer 2: ```suggestion
11. Dapp registers address at the Cast Server
```

4. Wallet derives symmetric key derived with keys X and Y
5. Push topic is derived from sha256 hash of symmetric key 
6. Wallet subscribes to push topic 
7. Wallet sends proposal response(type1 envelope) on pairing P with subscriptionAuth
8. Dapp receives proposal response on pairing P

> 9. Dapp derives symmetric key and dectypts subscriptionAuth

Reviewer 3: ```suggestion
9. Dapp derives symmetric key and decrypts subscriptionAuth
```
